### PR TITLE
Feature/improve firstboot xml

### PR DESCRIPTION
**Pull request by dgdavid**
## Problem

There are three clients/modules practically identical.

Related to

 * PRs #63, #66 
 * bsc [#1131327](https://bugzilla.suse.com/show_bug.cgi?id=1131327), [#1131301](https://bugzilla.suse.com/show_bug.cgi?id=1131301)

## Solution

Keep only one (_firstboot_licenses.rb_).

### How?

Just converting the rest of them (_firstboot_license.rb_ and _firstboot_license_novell.rb_) to symbolic links to ensure the backward compatibility.

In addition, in order to avoid showing the default license when only the selected one is desired, it is necessary to not take care about the default license directory since [this is already being done in the `InstLicense` client](https://github.com/yast/yast-installation/blob/aa816f10b7fa748eb7aafb9ecbacfbf9b2bab718/src/lib/installation/clients/inst_license.rb#L91-L110).

So, from now on, the firstboot.xml workflow only needs to include "firstboot_licenses" module

```xml
<module>
  <label>License Agreement</label>
  <enabled config:type="boolean">true</enabled>
  <name>firstboot_licenses</name>
</module>
```

Although using "firstboot_license" or "firstboot_novell_license" will produce the same results.

## Test

* Added a [basic unit test](https://github.com/yast/yast-firstboot/blob/ea2fe5f22f89a258df7c12bf34185409cd0c76fd/test/y2firstboot/clients/licenses_test.rb)
* Tested manually

## Screenshots

Let's see how the License agreement looks for each different scenarios. Click on each section to expand or collapse it.

<details>
<summary>Using the default values</summary>

---

That's, `FIRSTBOOT_LICENSE_DIR` and `FIRSTBOOT_LICENSE_NOVELL_DIR` are empty (`""`) or commented in _/etc/sysconfig/firstboot_.

The license texts found in the default directory (global param) will be displayed.

Screenshot_opensusetumbleweed_2019-05-10_15_19_38(1)

</details>

<details>
<summary>Using a valid path for FIRSTBOOT_LICENSE_DIR</summary>

---
Screenshot_opensusetumbleweed_2019-05-10_15_40_49(1)

Screenshot_opensusetumbleweed_2019-05-10_15_42_33

</details>

<details>
<summary>Using a valid path for FIRSTBOOT_LICENSE_NOVELL_DIR</summary>

---
Screenshot_opensusetumbleweed_2019-05-10_15_38_27(1)

Screenshot_opensusetumbleweed_2019-05-10_15_38_44

</details>

<details>
<summary>Using a valid path for both, FIRSTBOOT_LICENSE_DIR and FIRSTBOOT_LICENSE_NOVELL_DIR</summary>

---

Screenshot_opensusetumbleweed_2019-05-10_15_43_50(1)

Screenshot_opensusetumbleweed_2019-05-10_15_43_33

</details>

---

<small>NOTE: if any license file could not be found, the client simply will be skipped.</small>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,3 +26,5 @@ install-sh
 missing
 pluglib-bindings.ami
 *.pot
+coverage/
```
Comment (jreidinger) [inline .gitignore]: is this one needed? I think only one should be enough and also please use `/coverage` so it match only top level.

Comment (dgdavid) [reply .gitignore]: I just copied this values from the [`.gitignore` available in master](https://github.com/yast/yast-firstboot/blob/master/.gitignore) :)

But, for sure, we can improve it :+1:

File: src/lib/y2firstboot/clients/licenses.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    # valid path is given through FIRSTBOOT_NOVELL_LICENCE_DIR
+    class Licenses < Yast::Client
+      def initialize
+        textdomain "firstboot"
+      end
+
+      def run
+        result = WFM.CallFunction("inst_license", inst_license_args)
+
+        UI.CloseDialog if result == :halt
```
Comment (jreidinger) [inline src/lib/y2firstboot/clients/licenses.rb]: hmm, so every client have to close himself dialog if result is halt? cannot be done one level high?

Comment (dgdavid) [reply src/lib/y2firstboot/clients/licenses.rb]: Honestly, I don't know.

I tried to clean the client as much as possible, but this line I was not sure if could be changed. Maybe the firstboot module is only being closed if the user refuse the license.

File: test/test_helper.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    #     # => "be bigger than 2 and smaller than 4"
+    # ...rather than:
+    #     # => "be bigger than 2"
+    expectations.include_chain_clauses_in_custom_matcher_descriptions = true
+  end
+
+  config.mock_with :rspec do |mocks|
+    # Prevents you from mocking or stubbing a method that does not exist on
+    # a real object. This is generally recommended, and will default to
+    # `true` in RSpec 4.
```
Comment (jreidinger) [inline test/test_helper.rb]: I thought that it is default to true in rspec3, not?

File: test/test_helper.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+if ENV["COVERAGE"]
+  require "simplecov"
+  SimpleCov.start do
+    add_filter "/test/"
+  end
+
+  # for coverage we need to load all ruby files
+  src_location = File.expand_path("../../src", __FILE__)
+  Dir["#{src_location}/{module,lib}/**/*.rb"].each { |f| require_relative f }
```
Comment (jreidinger) [inline test/test_helper.rb]: ah, please no, this is old way to do this and cause some troubles. Now prefered way is https://github.com/yast/yast-bootloader/blob/master/test/test_helper.rb#L35 which also tracks all clients, so not anymore fake test coverage.

File: test/test_helper.rb (hunk 3 above)
Comment (dgdavid) [reply test/test_helper.rb]: No, it isn't.

At least as far as I can read :)

* https://github.com/rspec/rspec-core/blob/7b6b9c3f2e2878213f97d6fc9e9eb23c323cfe1c/lib/rspec/core/project_initializer/spec/spec_helper.rb

  > ... This is generally recommended, and will default to
# `true` in RSpec 4.

* https://github.com/rspec/rspec-mocks/blob/7dc2a431101172d31b780458eb168e05d20fb3aa/lib/rspec/mocks/configuration.rb

  > @verify_partial_doubles = false

File: src/lib/y2firstboot/clients/licenses.rb (hunk 2 above)
Comment (jreidinger) [reply src/lib/y2firstboot/clients/licenses.rb]: You cannot change it unless it is changed globally. So keep as it is now, maybe something to improve for future.

File: test/test_helper.rb (hunk 3 above)
Comment (jreidinger) [reply test/test_helper.rb]: yes, looks like. So I probably should enable it on more projects :)

**Pull request merged**